library/tendrl_user.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+module: test_user
+short_description: Creates user admin with password admin123 for tendrl.
+description:
+    - Creates user admin with password admin123 for tendrl
+    - There must be installed tendrl-api on the machine where this is ran.
+version_added: "1.0"
+author: "fbalak@redhat.com"
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import re
+import requests
+import os
+import socket
+import json
+
+hostname = socket.gethostname()
+module = AnsibleModule(argument_spec={})
+new_password = "admin123"
+
+
+def generate_user():
+    """Generates admin according to:
+    https://github.com/Tendrl/api/blob/master/docs/users.adoc#create-admin-user"""
+
+    os.chdir("/usr/share/tendrl-api")
+    return os.popen("RACK_ENV=production rake etcd:load_admin").read()
+
+
+def parse_password(string, module):
+    """Gets password from user_info. If there is none then it tries to login
+    with default password."""
+
+    password = re.compile("\"Password:\s*(?P<pass>[a-zA-Z0-9\._]+)\"")
+    admin_exists = re.compile("User named admin already exists.")
+    password_found = password.search(string)
+    admin_found = admin_exists.search(string)
+    if password_found is not None:
+        return password_found.group("pass")
+    elif admin_found is not None:
+        login("admin", new_password)
```

Comment:
Ah, this `login()` may end the whole process on failure. Could you drop a comment bout that?